src/s3path.jl

```diff
@@ -418,12 +417,13 @@ Base.ismount(fp::S3Path) = false
 """
     mkdir(fp::S3Path; recursive=false, exist_ok=false)
 
-Create an empty directory at the S3 path `fp`.  If `recursive`, this will create any previously non-existent
-directories which would contain `fp`.  An error will be thrown if an object exists at `fp` unless `exist_ok`.
-
-Note that empty directories in S3 are actually 0-byte objects with the naming convention of a directory.
+Create an empty directory within an existing bucket for the S3 path `fp`. If `recursive`,
+this will create any previously non-existent directories which would contain `fp`. An error
+will be thrown if an object exists at `fp` unless `exist_ok`.
```

Comment:
this would be pretty breaking but IMO the "recursive" case should actually be handled by `mkpath` instead of `mkdir`.

Reply:
I completely agree. I was going to do that as part of this PR but decided to keep that for another PR due to the breaking nature of the change.

Reply:
eh, if we're gonna make this breaking anyway due to the delimiter change, might as well do it all at once here.  or at least add the `mkpath` method and deprecate the recursive kwarg